Essentials/src/net/ess3/api/events/UserWarpEvent.java

```diff
@@ -0,0 +1,59 @@
+package net.ess3.api.events;
+
+import com.earth2me.essentials.Trade;
+import net.ess3.api.IUser;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+public class UserWarpEvent extends Event implements Cancellable {
+    private static final HandlerList handlers = new HandlerList();
+
+    private IUser user;
+    private String warp;
+    private Trade trade;
+    private boolean cancelled = false;
+
+
+    public UserWarpEvent(IUser user, String warp, Trade trade){
+        this.user = user;
+        this.warp = warp;
+        this.trade = trade;
+    }
+
+    public IUser getUser() {
+        return user;
+    }
+
+    public String getWarp() {
+        return warp;
+    }
+
+    public Trade getTrade() {
```

Comment:
Perhaps some javadocs would help?

Reply:
Should help. I am not that familiar with Javadocs tho

Reply:
I don't know if I were to left documentation there, providing that all of other Events doesn't have any documentation except the heading of the class.